### PR TITLE
loaders: deprecate cache loader

### PR DESCRIPTION
**Pull request by anshumanv**
Fixes https://github.com/webpack/webpack.js.org/issues/4144

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

**Comment by chenxsan** [comment]
As there's an unknown deployment issue, I've rebased the branch to the latest master branch to fix it, hope you're ok with it.

File: src/content/guides/build-performance.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ W> Don't use too many workers, as there is a boot overhead for the Node.js runti
 
 ### Persistent cache
 
-Enable persistent caching with the `cache-loader`. Clear cache directory on `"postinstall"` in `package.json`.
+Use [cache](https://webpack.js.org/configuration/other-options/#cache) option in webpack configuration. Clear cache directory on `"postinstall"` in `package.json`.
```
Comment (chenxsan) [inline src/content/guides/build-performance.md:108]:
```suggestion
Use [`cache`](/configuration/other-options/#cache) option in webpack configuration. Clear cache directory on `"postinstall"` in `package.json`.
```

**Comment by chenxsan** [comment]
Oops, failed again. I need to figure out what's going on with the Preview.

File: src/content/guides/build-performance.md (hunk 1 above)
Comment (anshumanv) [reply src/content/guides/build-performance.md]: thanks!

File: src/content/guides/build-performance.md [hunk 2]
```diff
@@ -105,7 +105,7 @@ W> Don't use too many workers, as there is a boot overhead for the Node.js runti
 
 ### Persistent cache
 
-Enable persistent caching with the `cache-loader`. Clear cache directory on `"postinstall"` in `package.json`.
+Use [`cache`](/configuration/other-options/#cache) option in webpack configuration. Clear cache directory on `"postinstall"` in `package.json`.
 
 
 T> We support yarn PnP version 3 [`yarn 2 berry`](https://next.yarnpkg.com/features/pnp) for persistent caching.
```
Comment (chenxsan) [inline src/content/guides/build-performance.md:111]:
```suggestion
T> We support yarn PnP version 3 [`yarn 2 berry`](https://yarnpkg.com/features/pnp) for persistent caching.
```

**Comment by chenxsan** [comment]
Just pushed another commit, and it fixed the Preview -- totally have no idea what's wrong with the vercel service.

**Review by chenxsan — APPROVED** [review]

**Comment by anshumanv** [comment]
Thanks!

**Pull request merged**